ttnn/cpp/ttnn/op_library/to_memory_config/to_memory_config_op.hpp

```diff
@@ -0,0 +1,158 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <functional>
+#include <optional>
+
+#include "tensor/tensor.hpp"
+// #include "third_party/magic_enum/magic_enum.hpp"
+// #include "tt_eager/tensor/owned_buffer_functions.hpp"
+// #include "tt_eager/tensor/tensor_utils.hpp"
+#include "tt_eager/tt_dnn/op_library/compute_kernel_config.hpp"
+#include "tt_eager/tt_dnn/op_library/copy/copy_op.hpp"
+#include "tt_eager/tt_dnn/op_library/run_operation.hpp"
+#include "tt_eager/tt_dnn/op_library/sharded/sharded_op.hpp"
+// #include "tt_metal/host_api.hpp"
+// #include "tt_metal/impl/dispatch/command_queue.hpp"
+// #include "ttnn/core.hpp"
+#include "ttnn/operations/core.hpp"
+#include "ttnn/types.hpp"
+
+namespace ttnn {
+
+namespace operations {
+
+namespace core {
+
+struct ToMemoryConfig {
+    static inline const std::array<TensorSchema, 1> input_tensor_schemas() {
+        return {ttnn::TensorSchema{
+            1,
+            4,
+            {ttnn::bfloat16, ttnn::bfloat8_b, ttnn::bfloat4_b, ttnn::float32, ttnn::uint16, ttnn::uint32, ttnn::int32},
+            {ttnn::ROW_MAJOR_LAYOUT, ttnn::TILE_LAYOUT},
+            true,
+            true,
+            false,
+            false}};
+    }
+
+    template <typename... Args>
+    static auto input_tensors_to_validate(const Tensor& tensor_arg, Args&&... args) {
+        return std::make_tuple(tensor_arg);
+    };
+
+    template <typename... Args>
+    static Tensor create_async_output_tensors(const ttnn::Tensor& tensor_arg, Args&&... args) {
+        return {{Tensor(operation::get_workers_for_op_output({tensor_arg}))}};
+    }
+
+    template <typename... Args>
+    static auto map_launch_op_args_to_execute(
+        const std::vector<Tensor>& input_tensors,
+        const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+        Args&&... args) {
+        return std::make_tuple(input_tensors.at(0), std::forward<Args>(args)...);
+    }
+
+    // TODO: Move to cpp once we merge with tt_eager
+    static Tensor execute(
+        const ttnn::Tensor tensor,
+        const ttnn::MemoryConfig& memory_config,
+        std::optional<ttnn::DataType> dtype) {
+        // Temporary until we see why buffer data not being populated
```

Comment:
move to .cpp?

Reply:
We have compilation issues right now with moving this to cpp. We will move it later